src/ecs/resources/input.rs

```diff
@@ -106,4 +155,45 @@ impl InputHandler {
     pub fn keys_once(&mut self, keys: &[VirtualKeyCode]) -> bool {
         keys.iter().any(|key| self.key_once(*key)) && self.keys_down(keys)
     }
+
+    /// Returns an iterator for all the pressed down mouse buttons.
+    pub fn pressed_mouse_buttons(&self) -> PressedMouseButtons {
+        PressedMouseButtons { iterator: self.pressed_mouse_buttons.keys() }
+    }
+
+    /// Checks if `button` is currently pressed.
+    pub fn mouse_button_down(&self, button: MouseButton) -> bool {
+        self.pressed_mouse_buttons.contains_key(&button)
+    }
+
+    /// Checks if all `buttons` are currently pressed.
+    pub fn mouse_buttons_down(&self, buttons: &[MouseButton]) -> bool {
+        buttons.iter().all(|btn| self.mouse_button_down(*btn))
+    }
+
+    /// Checks if the given mouse button is being pressed and held down.
+    ///
+    /// If `button` hasn't been let go since the last `mouse_button_once()` query, this
+    /// function will return false.
+    pub fn mouse_button_once(&mut self, button: MouseButton) -> bool {
+        if !self.pressed_mouse_buttons.contains_key(&button) {
+            return false;
+        }
+        if let Some(value) = self.pressed_mouse_buttons.get_mut(&button) {
+            // Should be safe
+            if *value == KeyQueryState::NotQueried {
+                *value = KeyQueryState::Queried;
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /// Checks if the all the given mouse buttons are being pressed and held down.
```

Comment:
In the situation where the mouse button(s) were pressed, then released, and not queried, then the result of the above comment would be true (i.e. they were pressend and not yet queried), but the function would return false (as they have been released).